File: app/src/main/java/com/example/myparking/HTTPRequests.java
package com.example.myparking;

import okhttp3.*;
import java.io.IOException;
import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.TimeUnit;

enum HttpMethods {
    GET,
    DELETE,
    POST,
    PUT
}

class HttpsRequests {
    private String appBaseUrl = "fe80::765d:2d84:f63d:9bcc%21:8081";
    private OkHttpClient client = new OkHttpClient().newBuilder()
            .connectTimeout(30, TimeUnit.SECONDS)
            .writeTimeout(30, TimeUnit.SECONDS)
            .readTimeout(30, TimeUnit.SECONDS)
            .build();

    private Request formRequest(String option, FormBody data, HttpMethods method) {
        switch (method) {
            case POST: return new Request.Builder().url(appBaseUrl + option).post(data).build();
            case PUT: return new Request.Builder().url(appBaseUrl + option).put(data).build();
            case DELETE: return new Request.Builder().url(appBaseUrl + option).delete(data).build();
            default: return new Request.Builder().url(appBaseUrl + option).get().build();
        }
    }

    // Sends request with given params
    String sendAsyncRequest(String option, HashMap<String, String> body, HttpMethods method) throws IOException {
        String result;

        FormBody.Builder postBody = new FormBody.Builder();
        for (Map.Entry<String, String> entry: body.entrySet()) {
            postBody.add(entry.getKey(), entry.getValue());
        }
        FormBody postData = postBody.build();

        Request request = formRequest(option, postData, method);

        Response resp = client.newCall(request).execute();
        return (resp.isSuccessful()) ? resp.body().string() : "";
    }
}

File: app/src/main/java/com/example/myparking/HttpRequests.kt
package com.example.myparking

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkCapabilities
//import com.example.simple_english.data.Constants
//import com.example.simple_english.data.HttpMethods
//import kotlinx.serialization.json.*
import okhttp3.*
import okio.IOException
import java.net.URLEncoder
import java.util.concurrent.TimeUnit

enum class HttpMethods {
    GET,
    POST,
    PUT
}

class HttpsRequests {
    private val appBaseUrl = "http://localhost:8081"
    private val client = OkHttpClient().newBuilder()
            .connectTimeout(30, TimeUnit.SECONDS)
            .writeTimeout(30, TimeUnit.SECONDS)
            .readTimeout(30, TimeUnit.SECONDS)
            .build()

    private fun formRequest(option: String, data: FormBody, method: HttpMethods): Request =
            when (method) {
                HttpMethods.POST -> Request.Builder().url(appBaseUrl + option).post(data).build()
                HttpMethods.PUT -> Request.Builder().url(appBaseUrl + option).put(data).build()
                HttpMethods.GET -> Request.Builder().url(appBaseUrl + option).get().build()
            }

    // Sends request with given params
    fun sendAsyncRequest(
            option: String,
            body: Map<String, String>,
            method: HttpMethods
    ): String {
        val result: String

        val postBody = FormBody.Builder()
        for ((key, value) in body) {
            postBody.add(key, value)
        }
        val postData = postBody.build()

        val request = formRequest(option, postData, method)

        client.newCall(request).execute().use { response ->
            result = when (response.isSuccessful) {
                true -> response.body!!.string()
                false -> ""
            }
        }

        return result
    }
}